src/App.ts

```diff
@@ -0,0 +1,118 @@
+import * as net from 'net';
+const fs = require('fs');
+const electron = require('electron');
+const path = require('path');
+const url = require('url');
+
+
+export class App
+{
+
+    public listen()
+    {
+        let num: number = 0;
+        let path: string = './Downloads';
+        var exist = this.statPath(path);
+        if(!exist)
+        {
+            fs.mkdir(path);
+        }
+        electron.app.on('ready', this.createWindow);
+
+        electron.ipcMain.on('async-port', (event, port) =>
+        {
+            const server = net.createServer((socket) =>
+            {
+
+                socket.on('end', () =>
+                {
+                    console.log('client disconnected');
+                });
+
+                socket.on('data',(data) =>
+                {
+                    console.log('Receiving data\nTransfered Bytes : ' + data.length);
+                    let fileToSave = path + '/file' + num;
+                    num++;
+                    console.log(fileToSave);
+                    fs.open(fileToSave, "w", function (err, fd) 
+                    {
+                        fs.writeFile(fileToSave, data, (err) =>
+                        {
+                            if (err) throw err;
+                            console.log('The file has been saved!');
+                        });
+                    });
+                });
+            });
+
+            server.listen(port, () =>
+            {
+                console.log('server is listening on port ' + port);
+            });
+
+            server.on('error', (err) =>
+            {
+                throw err;
+            });
+    
+            server.on('connection',() =>
+            {
+                console.log('Connection to client established');
+            });
+
+        });
+
+        electron.ipcMain.on('async', (event, ip, port, file) =>
+        {
+            var socket = new net.Socket;
```

Comment:
Var? Please avoid using it. Is this working? Are you sure? I think it should be `let socket = new net.Socket();`